From 64b2c8524474fb0e5f9aeca64f0d6623cb9f3edf Mon Sep 17 00:00:00 2001
From: "sai.shwethas" <sai.shwethas@nxp.com>
Date: Mon, 2 Sep 2024 17:10:18 +0530
Subject: [PATCH 01/18] [artf1133064]: Fix for NFC Forum T5T testcase failures

Added config SKIP_ISO15693_GET_SYS_INFO_CMD config option to skip
ISO15693 GET_SYS_INFO command as NFC forum tool does not support
this command.

Change-Id: I3d911c0e8018407680eed83c534ec46cd509de84
---
 .../halimpl/conf/PN557/gen-config-files/libnfc-nci.conf  | 6 ++++++
 .../halimpl/conf/PN560/gen-config-files/libnfc-nci.conf  | 8 +++++++-
 .../conf/SN1xx/sn100/gen-config-files/libnfc-nci.conf    | 9 +++++++--
 .../conf/SN1xx/sn110/gen-config-files/libnfc-nci.conf    | 8 +++++++-
 .../halimpl/conf/SN220/gen-config-files/libnfc-nci.conf  | 9 +++++++--
 .../halimpl/conf/SN300/gen-config-files/libnfc-nci.conf  | 9 +++++++--
 6 files changed, 41 insertions(+), 8 deletions(-)

diff --git a/snxxx/halimpl/conf/PN557/gen-config-files/libnfc-nci.conf b/snxxx/halimpl/conf/PN557/gen-config-files/libnfc-nci.conf
index 6bd7aebf..e5526f5d 100644
--- a/snxxx/halimpl/conf/PN557/gen-config-files/libnfc-nci.conf
+++ b/snxxx/halimpl/conf/PN557/gen-config-files/libnfc-nci.conf
@@ -120,3 +120,9 @@ PRESENCE_CHECK_RETRY_COUNT=0
 # 0x01, Remove mute techs from rf discover cmd
 MUTE_TECH_ROUTE_OPTION=0x01
 ###############################################################################
+# Config option to skip ISO15693 GET_SYS_INFO command as NFC forum tool does
+# not support this command
+# 0 to Disable this behaviour
+# 1 to Enable this behaviour
+ISO15693_SKIP_GET_SYS_INFO_CMD=0
+##############################################################################
diff --git a/snxxx/halimpl/conf/PN560/gen-config-files/libnfc-nci.conf b/snxxx/halimpl/conf/PN560/gen-config-files/libnfc-nci.conf
index 19637db6..968be4dc 100644
--- a/snxxx/halimpl/conf/PN560/gen-config-files/libnfc-nci.conf
+++ b/snxxx/halimpl/conf/PN560/gen-config-files/libnfc-nci.conf
@@ -92,5 +92,11 @@ PRESENCE_CHECK_RETRY_COUNT=0
 # 0x00, Route mute techs to DH, enable block bit and set power state to 0x00
 # 0x01, Remove mute techs from rf discover cmd
 MUTE_TECH_ROUTE_OPTION=0x01
-#########################################################################
+##############################################################################
+# Config option to skip ISO15693 GET_SYS_INFO command as NFC forum tool does
+# not support this command
+# 0 to Disable this behaviour
+# 1 to Enable this behaviour
+ISO15693_SKIP_GET_SYS_INFO_CMD=0
+##############################################################################
 
diff --git a/snxxx/halimpl/conf/SN1xx/sn100/gen-config-files/libnfc-nci.conf b/snxxx/halimpl/conf/SN1xx/sn100/gen-config-files/libnfc-nci.conf
index 2e7ca676..ecd827c7 100644
--- a/snxxx/halimpl/conf/SN1xx/sn100/gen-config-files/libnfc-nci.conf
+++ b/snxxx/halimpl/conf/SN1xx/sn100/gen-config-files/libnfc-nci.conf
@@ -99,5 +99,10 @@ PRESENCE_CHECK_RETRY_COUNT=0
 # 0x00, Route mute techs to DH, enable block bit and set power state to 0x00
 # 0x01, Remove mute techs from rf discover cmd
 MUTE_TECH_ROUTE_OPTION=0x01
-#########################################################################
-
+##############################################################################
+# Config option to skip ISO15693 GET_SYS_INFO command as NFC forum tool does
+# not support this command
+# 0 to Disable this behaviour
+# 1 to Enable this behaviour
+ISO15693_SKIP_GET_SYS_INFO_CMD=0
+##############################################################################
diff --git a/snxxx/halimpl/conf/SN1xx/sn110/gen-config-files/libnfc-nci.conf b/snxxx/halimpl/conf/SN1xx/sn110/gen-config-files/libnfc-nci.conf
index 2e7ca676..81729393 100644
--- a/snxxx/halimpl/conf/SN1xx/sn110/gen-config-files/libnfc-nci.conf
+++ b/snxxx/halimpl/conf/SN1xx/sn110/gen-config-files/libnfc-nci.conf
@@ -99,5 +99,11 @@ PRESENCE_CHECK_RETRY_COUNT=0
 # 0x00, Route mute techs to DH, enable block bit and set power state to 0x00
 # 0x01, Remove mute techs from rf discover cmd
 MUTE_TECH_ROUTE_OPTION=0x01
-#########################################################################
+##############################################################################
+# Config option to skip ISO15693 GET_SYS_INFO command as NFC forum tool does
+# not support this command
+# 0 to Disable this behaviour
+# 1 to Enable this behaviour
+ISO15693_SKIP_GET_SYS_INFO_CMD=0
+##############################################################################
 
diff --git a/snxxx/halimpl/conf/SN220/gen-config-files/libnfc-nci.conf b/snxxx/halimpl/conf/SN220/gen-config-files/libnfc-nci.conf
index f61c0317..f278dc0e 100644
--- a/snxxx/halimpl/conf/SN220/gen-config-files/libnfc-nci.conf
+++ b/snxxx/halimpl/conf/SN220/gen-config-files/libnfc-nci.conf
@@ -100,5 +100,10 @@ PRESENCE_CHECK_RETRY_COUNT=0
 # 0x00, Route mute techs to DH, enable block bit and set power state to 0x00
 # 0x01, Remove mute techs from rf discover cmd
 MUTE_TECH_ROUTE_OPTION=0x01
-#########################################################################
-
+##############################################################################
+# Config option to skip ISO15693 GET_SYS_INFO command as NFC forum tool does
+# not support this command
+# 0 to Disable this behaviour
+# 1 to Enable this behaviour
+ISO15693_SKIP_GET_SYS_INFO_CMD=0
+##############################################################################
diff --git a/snxxx/halimpl/conf/SN300/gen-config-files/libnfc-nci.conf b/snxxx/halimpl/conf/SN300/gen-config-files/libnfc-nci.conf
index 2964063b..7f822f47 100644
--- a/snxxx/halimpl/conf/SN300/gen-config-files/libnfc-nci.conf
+++ b/snxxx/halimpl/conf/SN300/gen-config-files/libnfc-nci.conf
@@ -106,5 +106,10 @@ PRESENCE_CHECK_RETRY_COUNT=0
 # 0x00, Route mute techs to DH, enable block bit and set power state to 0x00
 # 0x01, Remove mute techs from rf discover cmd
 MUTE_TECH_ROUTE_OPTION=0x01
-#########################################################################
-
+##############################################################################
+# Config option to skip ISO15693 GET_SYS_INFO command as NFC forum tool does
+# not support this command
+# 0 to Disable this behaviour
+# 1 to Enable this behaviour
+ISO15693_SKIP_GET_SYS_INFO_CMD=0
+##############################################################################
\ No newline at end of file
-- 
2.47.0

